src/pages/api/newsletter-signup.ts

```diff
@@ -0,0 +1,35 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import mailchimp from '@mailchimp/mailchimp_marketing';
+
+mailchimp.setConfig({
+  apiKey: process.env.MAILCHIMP_APIKEY,
+  server: process.env.MAILCHIMP_SERVER
+});
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { body, method } = req;
+
+  if (method !== 'POST') {
```

Comment:
Is this necessary?

Reply:
Hmm not totally necessary I would say, just a guard check, to fail quickly. Let me know if you feel this is not adding much value and I can remove it.

Reply:
Not sure, I've never use a guard like that to be honest, don't know if the method can be 'hijacked' someway. What do you think @samajammin?